frontend/src/components/beta/common/NFTMedia.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from "react";
+import { Img } from "react-image";
+import { styled } from "../../../libraries/styles";
+import PaginationSpinner from "../../utils/PaginationSpinner";
+
+const FallbackImage = styled("img", {
+  width: "100%",
+});
+
+type Props = {
+  src: string | null;
+};
+const NFTMedia: React.FC<Props> = React.memo(({ src }) => {
+  const [mediaUrl, setMediaUrl] = React.useState(src ?? null);
+  const handleMediaUrl = React.useCallback(
+    () => setMediaUrl("/static/images/failed_to_load.svg"),
+    [setMediaUrl]
+  );
+  const [poster, setPoster] = React.useState("");
+  const handlePoster = React.useCallback(
+    () => setPoster("/static/images/failed_to_load.svg"),
+    [setPoster]
+  );
+
+  if (!mediaUrl) {
+    return <FallbackImage src="/static/images/failed_to_load.svg" />;
+  }
+  // copied from wallet https://github.com/near/near-wallet/blob/master/packages/frontend/src/components/nft/NFTMedia.js
+  const [isVideo, mimeType] = React.useMemo(() => {
+    let mimeType;
+    // check mediaUrl string for .webm or .mp4 endings (case-insensitive)
+    if (mediaUrl.match(/\.webm$/i)) mimeType = "webm";
+    else if (mediaUrl.match(/\.mp4$/i)) mimeType = "mp4";
+    // if there is a mediaUrl and a truthy mimeType (webm or mp4), we have a video
+    const isVideo = !!mediaUrl && mimeType;
+    return [isVideo, mimeType];
+  }, [mediaUrl]);
+
+  return (
+    <>
+      {isVideo ? (
+        <video muted={true} loop controls autoPlay={false} poster={poster}>
+          <source
+            src={mediaUrl}
+            type={`video/${mimeType}`}
+            onError={handlePoster}
+          />
+        </video>
+      ) : (
+        <Img
```

Comment:
I'd like to use nextjs/image `<Image />` component, but using that component requires a list of domains which could be infinite. And we can't set it as `dynamic` or something like that.

Reply:
There is no reason to make `<img>` dynamic as there is no heavy rendering behind getting `<img>` rendered to HTML output on the server side. Why do you want to use `<Image>` over `<img>`? I feel `<Image>` might cache the image on build which is not the behavior we need here as NFT images are not known ahead of time and we don't want to pull them in to be part of the project (they are external resources).

Reply:
I wanted to use it because nextjs/image could compress images on the fly, that's why the images could load much faster

Reply:
There is a reason to have a list of allowed domains - to prevent attacks from malicious users.
I guess you can use proxy servers for that: https://nextjs.org/docs/api-reference/next/image#loader-configuration